met-api/src/met_api/models/event_item.py

```diff
@@ -29,3 +31,36 @@ class EventItem(BaseModel):  # pylint: disable=too-few-public-methods, too-many-
     def save_event_items(cls, event_items: list) -> None:
         """Update widgets.."""
         db.session.bulk_save_objects(event_items)
+        
+    
+    @classmethod
+    def update_event_item(cls, event_data: dict) -> Optional[EventItem or DefaultMethodResult]:
+        """Update Event Item."""
+        query = EventItem.query.filter_by(id=event_data.get('id'))
+        update_fields = dict(   
```

Comment:
If we're patching we don't want to ideally touch all fields when we update, we can pass only the affected fields. We can follow the pattern in the engagement model. edit_engagment is what we did so we don't do it the update_engagement way and send all fields to the update
![image](https://user-images.githubusercontent.com/91914654/221994569-de6080e5-53d8-45e7-bcc3-5550b58f81e2.png)

Reply:
good idea there